src/main/java/com/linkedin/kmf/services/TopicManagementServiceManager.java

```diff
@@ -0,0 +1,154 @@
+ */
+
+package com.linkedin.kmf.services;
+
+import com.linkedin.kmf.apps.configs.MirrorPipelineMonitorConfig;
+import com.linkedin.kmf.services.configs.CommonServiceConfig;
+import com.linkedin.kmf.services.configs.ConsumeServiceConfig;
+import com.linkedin.kmf.services.configs.TopicManagementServiceConfig;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class TopicManagementServiceManager implements Service {
+  private static final Logger LOG = LoggerFactory.getLogger(TopicManagementServiceManager.class);
+
+  private double _partitionsToBrokerRatio;
+  private String _name;
+  private List<String> _zookeeperUrls;
+  private AtomicBoolean _running;
+  private final ScheduledExecutorService _managementExecutor;
+  private Map<String, TopicManagementService> _services;
+
+  public TopicManagementServiceManager(Map<String, Object> props, List<String> zookeeperUrls, String topic,
+                                        String name) throws Exception {
+    Map<String, Object> topicManagementProps = createTopicManagementServiceProps(props, topic);
+    topicManagementProps.put(CommonServiceConfig.ZOOKEEPER_CONNECT_CONFIG, "");
+    TopicManagementServiceConfig config = new TopicManagementServiceConfig(topicManagementProps);
+    _partitionsToBrokerRatio = config.getDouble(TopicManagementServiceConfig.PARTITIONS_TO_BROKER_RATIO_THRESHOLD);
+    _name = name;
+    _zookeeperUrls = zookeeperUrls;
+    _running = new AtomicBoolean(false);
+    _managementExecutor = Executors.newSingleThreadScheduledExecutor(new TopicManagementServiceManagerThreadFactory());
+    Map<String, Integer> brokerCounts = getBrokerCounts(zookeeperUrls);
+    int maxNumPartitions = getMaxNumberOfPartitions(brokerCounts);
+
+    _services = new HashMap<>();
+    for (String zookeeper: _zookeeperUrls) {
+      topicManagementProps.put(ConsumeServiceConfig.ZOOKEEPER_CONNECT_CONFIG, zookeeper);
+      double individualPartitionToBrokerRatio = maxNumPartitions / brokerCounts.get(zookeeper);
+      topicManagementProps.put(TopicManagementServiceConfig.PARTITIONS_TO_BROKER_RATIO_CONFIG, individualPartitionToBrokerRatio);
+      _services.put(zookeeper, new TopicManagementService(topicManagementProps, name + ":" + zookeeper));
+    }
+
+    LOG.info(_name + ": topic management service manager is initialized.");
+  }
+
+  @SuppressWarnings("unchecked")
+  private Map<String, Object> createTopicManagementServiceProps(Map<String, Object> props, String topic) {
+    Map<String, Object> topicManagementServiceProps = props.containsKey(MirrorPipelineMonitorConfig.TOPIC_MANAGEMENT_SERVICE_CONFIG)
```

Comment:
We want each service to be reusable. App can depend on Service's config but not the vise versa. Thus we don't want a service class to reference a config from MirrorPipelineMonitorConfig.